MetaMask/metamask-extension
# Fix + Clean: "Security & Privacy" Settings Search routes

## **Description**

- Fix search for:
  - Network Details Check
  - Proposed Names
  - Show balance and token price checker
  - Use Phishing Detection
- Improve search for: 
  - "Choose your network"
    - Previously, did not show when typing "Net.." or "Network" 
    - Previously only showed when typing "Cho..." 
    - Searching "Net..." or "Cho..." will now redirect to "Network Provider"
- Clean:
  - Add some JSDoc comments to assist understanding settings search logic
  - Rename SettingsSearch function getNumberOfSettingsInSection -> getNumberOfSettingRoutesInTab
    - Previously it was a misleading name
      - not of section; it is of the tab
      - not counting all settings in tabs, only counting routed settings
  - Tests
    - update SettingsSearch#getSettingsRoutes
    - update test phrases 
  
Mismatching references caused some searches to lead to the wrong sections. 
note: not attempting to cleanup everything. Just helping out a bit while I pass by this part of the code. 

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/22985

## **Manual testing steps**

1. Go to settings
2. Search one of the examples in the related PR
3. Observe scroll behavior

## **Screenshots/Recordings**



### **Before**

<img width="520" alt="Screenshot 2024-02-15 at 7 00 03 PM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/a8d302e7-cd32-46c4-8046-0dbeca9c4993">
<img width="520" alt="Screenshot 2024-02-15 at 7 00 17 PM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/a0602b57-8275-4c26-a5be-6ec1f584ccc2">


### **After**

<img width="520" alt="Screenshot 2024-02-15 at 7 03 18 PM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/12be06bd-b726-4c1a-9c13-bcbf80997e66">
<img width="520" alt="Screenshot 2024-02-15 at 7 03 28 PM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/0bc3596a-9264-48a3-ab5c-9c981e04e9ef">

(now scrolls to "Network Provider" instead of cutting "Network Provider" off)
<img width="520" alt="Screenshot 2024-02-15 at 7 11 32 PM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/d62a2fc5-b761-45b2-a223-daa5d06f7798">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
